meson.build

```diff
@@ -0,0 +1,147 @@
+
+targets = get_option('target')
+if 'auto' in targets or 'all' in targets
+  if can_target_aarch64 or 'all' in targets
+    add_project_arguments(
+      '-DVIXL_INCLUDE_TARGET_A32',
+      '-DVIXL_INCLUDE_TARGET_T32',
+      '-DVIXL_INCLUDE_TARGET_A64',
+      language: 'cpp'
+    )
+    build_a32 = true
+    build_t32 = true
+    build_a64 = true
+  else
+    add_project_arguments(
+      '-DVIXL_INCLUDE_TARGET_A32',
+      '-DVIXL_INCLUDE_TARGET_T32',
+      language: 'cpp'
+    )
+    build_a32 = true
+    build_t32 = true
+  endif
+else
+  if 'a32' in targets or 'aarch32' in targets
+    add_project_arguments(
+      '-DVIXL_INCLUDE_TARGET_A32',
+      language: 'cpp'
+    )
+    build_a32 = true
+  endif
+  if 't32' in targets or 'aarch32' in targets
+    add_project_arguments(
+      '-DVIXL_INCLUDE_TARGET_T32',
+      language: 'cpp'
+    )
+    build_t32 = true
+  endif
+  if 'a64' in targets or 'aarch64' in targets
+    add_project_arguments(
+      '-DVIXL_INCLUDE_TARGET_A64',
+      language: 'cpp'
+    )
+    build_a64 = true
+  endif
+endif
+
+target_sources = []
+if build_a32 or build_t32
+  subdir('src'/'aarch32')
+endif
+if build_a64
+  subdir('src'/'aarch64')
+endif
+
+if get_option('simulator') == 'auto'
+  if not (host_machine.cpu_family() == 'aarch64') and can_target_aarch64
+    add_project_arguments('-DVIXL_INCLUDE_SIMULATOR_AARCH64', language: 'cpp')
+  endif
+elif get_option('simulator') == 'aarch64'
+  if can_target_aarch64 and build_a64
+    add_project_arguments('-DVIXL_INCLUDE_SIMULATOR_AARCH64', language: 'cpp')
+  else
+    error('Building an AArch64 simulator implies that VIXL targets AArch64. Set `target` to include `aarch64` or `a64`.')
+  endif
+endif
+
+allocator = get_option('code_buffer_allocator')
+if (allocator == 'auto' and host_machine.system() == 'linux') or allocator == 'mmap'
+  add_project_arguments('-DVIXL_CODE_BUFFER_MMAP', language: 'cpp')
+else
+  add_project_arguments('-DVIXL_CODE_BUFFER_MALLOC', language: 'cpp')
+endif
+
+markdown = find_program('markdown', required: get_option('doc'))
+if markdown.found()
+  subdir('doc')
+endif
+
+libvixl = library(
+  'vixl',
+  'src'/'code-buffer-vixl.cc',
+  'src'/'compiler-intrinsics-vixl.cc',
+  'src'/'cpu-features.cc',
+  'src'/'utils-vixl.cc',
+  sources: target_sources,
+  include_directories: 'src',
+  version: meson.project_version(),
+  install: true
+)
+
+vixl_dep = declare_dependency(link_with: libvixl, include_directories: 'src')
+
+if meson.version().version_compare('>=0.54.0')
+  meson.override_dependency('vixl', vixl_dep)
+endif
+
+install_headers(
+  'src'/'assembler-base-vixl.h',
+  'src'/'code-buffer-vixl.h',
+  'src'/'code-generation-scopes-vixl.h',
+  'src'/'compiler-intrinsics-vixl.h',
+  'src'/'cpu-features.h',
+  'src'/'globals-vixl.h',
+  'src'/'invalset-vixl.h',
+  'src'/'macro-assembler-interface.h',
+  'src'/'platform-vixl.h',
+  'src'/'pool-manager-impl.h',
+  'src'/'pool-manager.h',
+  'src'/'utils-vixl.h',
+  subdir: 'vixl'
```

Comment:
These files are installed to e.g. `/usr/include/vixl/utils-vixl.h`, but immediately below you generate a pkg-config file that doesn't specify `subdirs: vixl`.

So, the assumption seems to be that people compile against these headers using `#include <vixl/utils-vixl.h>` etc.
But then above you set up a declared dependency adding `src/` to the in-tree include path, which only works if includes are done in the style `#include <utils-vixl.h>`.

This is inconsistent and probably not what you meant. :)

Reply:
Thank you for your reviewing this, I'll fix the issues today :)

Reply:
Fixed. I chose to add `vixl` as an include subdirectory of the pkg-config file, as the examples include files like `#include <utils-vixl.h>`